File: OP2_labs/lab4_cpp/func.cpp
#pragma once
#include "Header.h"
#include "Teacher.h"
#include "Student.h"

void readFile(string filename, vector<Teacher> &teachers, vector<Student> &students)
{
	ifstream infile;
	infile.open(filename);
	vector<string> person;
	string row;
	while (getline(infile, row))
	{
		if ((row[1] == ' ') && ((row[0] == 'S') || (row[0] == 'T'))) {
			if (person.size() != 0)
			{
				if (person[0][0] == 'S')
				{
					students.push_back(Student(person));
					person.clear();
				}
				else if (person[0][0] == 'T')
				{
					teachers.push_back(Teacher(person));
					person.clear();
				}
			}
		}
		person.push_back(row);
	}
	if (person.size() != 0)
	{
		if (person[0][0] == 'S')
		{
			students.push_back(Student(person));
			person.clear();
		}
		else if (person[0][0] == 'T')
		{
			teachers.push_back(Teacher(person));
			person.clear();
		}
	}
	infile.close();
}

void outAvgTeacherPopularity(vector<Teacher> teachers)
{
	cout << "Calculating average popularity of these teachers: \n";
	float curr, all = 0, avg;
	for (Teacher teacher : teachers)
	{
		curr = teacher.getAvgPopularity();
		all += curr;
		cout << teacher.getName() << ' ' << curr << '\n';
	}
	avg = all / teachers.size();
	cout << "Average popularity: " << avg << "\n\n";
}

void outAvgStudentGrade(vector<Student> students)
{
	cout << "Calculating average grades of these students: \n";
	float curr, all = 0, avg;
	for (Student student : students)
	{
		curr = student.getAvgGrade();
		all += curr;
		cout << student.getName() << ' ' << curr << '\n';
	}
	avg = all / students.size();
	cout << "Average grades: " << avg << "\n\n";
}

void outMinorAcdDebtors(vector<Student> students)
{
	int count = 0;
	for (Student student : students)
	{
		if ((!student.isAdult()) && (student.hasAcdDebt())) { count++; }
	}
	cout << "less than 18 year old students with academic debt: " << count << '\n';
}

File: OP2_labs/lab4_py/func.py
from Student import *
from Teacher import *

def readFile(filename, students, teachers):
    infile = open(filename, 'r')
    text = infile.read().split('\n')
    person = []
    for row in text:
        if (row[1] == ' ') and ((row[0] == 'S') or (row[0] == 'T')):
            if len(person) != 0:
                if person[0][0] == 'S':
                    students.append(Student(person))
                    person.clear()
                elif person[0][0] == 'T':
                    teachers.append(Teacher(person))
                    person.clear()
        person.append(row)
    if len(person) != 0:
        if person[0][0] == 'S':
            students.append(Student(person))
            person.clear()
        elif person[0][0] == 'T':
            teachers.append(Teacher(person))
            person.clear()
    infile.close()

def outAvgTeacherPopularity(teachers):
    print("Calculating average popularity of these teachers:")
    curr : float
    all : float = 0.0
    avg : float
    for teacher in teachers:
        curr = teacher.getAvgPopularity()
        all += curr
        print(teacher.getName(), curr)
    avg = all / len(teachers)
    print("Average popularity: ", avg, "\n")

def outAvgStudentGrade(students):
    print("Calculating average grades of these students: \n")
    curr : float
    all : float = 0.0
    avg : float
    for student in students:
        curr = student.getAvgGrade()
        all += curr
        print(student.getName(), curr)
    avg = all / len(students)
    print("Average grades: ", avg, "\n")

def outMinorAcdDebtors(students):
    count = 0
    for student in students:
        if (not student.isAdult()) and (student.hasAcdDebt()):
            count += 1
    print("less than 18 year old students with academic debt: ", count)